docs/source/tutorial/performance.rst

```diff
@@ -0,0 +1,115 @@
+performing link prediction tasks the goal is to score all entities/relations for a given tuple, i.e. (h, r,), (r, t) or
+(h, t). In these cases a single tuple is used many times for different entities/relations.
+
+E.g. we want to rank all entities for a single tuple (h, r) with the DistMult model for the Freebase15k237 dataset. This
+dataset contains 14,505 entites, which means that we have 14,505 (h, r, t) combinations, whereas _h_ and _r_ are
+constant. Looking at the interaction function of the DistMult model, we can observe that the :math:`h*r` part causes
+half of the mathematical operations to calculate :math:`h*r*t`. Therefore, calculating the :math:`h*r` part only once
+and reusing it spares us half of the mathematical operations for the other 14,504 remaining entities, making the
+calculations roughly twice as fast in total. The speed-up might be significantly higher in cases where the broadcasted
+part has a high relative complexity compared to the overall interaction function, e.g. :class:`pykeen.models.ConvE`.
+
+To make this technique possible, PyKEEN models have to provide an explicit broadcasting function via following methods
+in the model class:
+ - :func:`score_h` - Scoring all possible head entities for a given (r, t) tuple
+ - :func:`score_r` - Scoring all possible relations for a given (h, t) tuple
+ - :func:`score_t` - Scoring all possible tail entities for a given (h, r) tuple
+The PyKEEN architecture natively supports these methods and makes use of this technique wherever possible without any
+additional modifications. Providing these methods is completely optional and not required when implementing new models.
+
+Filtering with index-based masking
+----------------------------------
+In a standard evaluation setting of a KGEM for each triple (h, r, t) in the test/validation dataset two calculations are
+performed:
+ - the tuple (h, r) is combined with all possible tail entities t*
+ - the tuple (r, t) is combined with all possible head entities h*
+Afterwards the rank of (h, r, t) is compared to all possible (h, r, t*) as well as (h*, r, t) triples.
+
+In the filtered setting, t* is not allowed to contain tail entities that would lead to (h, r, t*) triples already found
+in the train dataset. Analogue to that, h* is not allowed to contain head entities leading to (h*, r, t) triples found
+in the train dataset. This leads to the computational challenge that all new possible triples (h, r, t*) and (h*, r, t)
+have to be checked against their existence in the train dataset. Considering a dataset like FB15k237, with almost 15,000
+entities, each test triples leads to 30,000 possible new triples, which have to be checked against the train dataset.
+After removing all possible entities found in the train dataset from h* and t*, new sets h** and t** are obtained that
+allow to construct purely novel triples (h**, r, t) and (h, r, t**) not found in the train dataset.
```

Comment:
@lvermue I think we have to put the `**` part into math-mode - otherwise it only triggers bold-font.